### PR TITLE
chore(bullmq-task-queue): upgrade `bullmq` to latest

### PR DESCRIPTION
**Pull request by yaanno**
This PR contains some module updates and code changes as follows. 

BullMQ changes deprecated the QueueScheduler alltogether for new, simplified APIs. Since this example is a simple queue/worker setup (without Flows), the scheduler setup has been removed.

Major versions:
- BullMQ: 1.7 -> 2 (Breaking change, see `queue.server.ts` fix)

Minor versions:
- IORedis

### CODE REVIEW

File: bullmq-task-queue/app/entry.client.tsx [hunk 1]
```diff
@@ -1,4 +1,21 @@
+import { startTransition, StrictMode } from "react";
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/app/entry.client.tsx:1]:
```suggestion
```

File: bullmq-task-queue/app/entry.client.tsx [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
 import { RemixBrowser } from "@remix-run/react";
-import { hydrate } from "react-dom";
+import { hydrateRoot } from "react-dom/client";
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/app/entry.client.tsx:3]:
```suggestion
import { hydrate } from "react-dom";
```

File: bullmq-task-queue/app/entry.client.tsx [hunk 3]
```diff
@@ -1,4 +1,21 @@
...
+    );
+  });
+};
+if (window.requestIdleCallback) {
+  window.requestIdleCallback(hydrate);
+} else {
+  // Safari doesn't support requestIdleCallback
+  // https://caniuse.com/requestidlecallback
+  window.setTimeout(hydrate, 1);
+}
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/app/entry.client.tsx:21]:
```suggestion
hydrate(<RemixBrowser />, document);
```

File: bullmq-task-queue/package.json [hunk 4]
```diff
@@ -10,19 +10,19 @@
...
     "@remix-run/react": "*",
     "@remix-run/serve": "*",
-    "bullmq": "^1.76.0",
-    "ioredis": "^4.28.5",
-    "react": "^17.0.2",
-    "react-dom": "^17.0.2"
+    "bullmq": "^2.1.0",
+    "ioredis": "^5.2.3",
+    "react": "^18.2.0",
+    "react-dom": "^18.2.0"
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/package.json:16]:
```suggestion
    "react": "^17.0.2",
    "react-dom": "^17.0.2"
```

File: bullmq-task-queue/package.json [hunk 5]
```diff
@@ -10,19 +10,19 @@
...
     "@remix-run/dev": "*",
     "@remix-run/eslint-config": "*",
-    "@types/ioredis": "^4.28.8",
-    "@types/react": "^17.0.39",
-    "@types/react-dom": "^17.0.13",
-    "eslint": "^8.10.0",
-    "typescript": "^4.7.4"
+    "@types/ioredis": "^4.28.10",
+    "@types/react": "^18.0.21",
+    "@types/react-dom": "^18.0.6",
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/package.json:23]:
```suggestion
    "@types/react": "^17.0.39",
    "@types/react-dom": "^17.0.13",
```

File: bullmq-task-queue/package.json [hunk 6]
```diff
@@ -10,19 +10,19 @@
...
+    "eslint": "^8.24.0",
+    "typescript": "^4.8.3"
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/package.json:25]: Let's keep these the same as well, to be consistent with all other examples

```suggestion
    "eslint": "^8.10.0",
    "typescript": "^4.7.4"
```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]
Thanks for working on this! 🙏

Could you please revert all React 18 changes?
We're going to update all examples to React 18 once our default templates are updated as well, but we're still figuring some things out there.

**Comment by yaanno** [comment]
> Thanks for working on this! 🙏
> 
> Could you please revert all React 18 changes? We're going to update all examples to React 18 once our default templates are updated as well, but we're still figuring some things out there.

Sure, no problem. Makes sense to do the React upgrade on one move.

File: bullmq-task-queue/app/entry.client.tsx [hunk 7]
```diff
@@ -1,4 +1,4 @@
 import { RemixBrowser } from "@remix-run/react";
 import { hydrate } from "react-dom";
 
-hydrate(<RemixBrowser />, document);
+hydrate(<RemixBrowser />, document);
```
Comment (MichaelDeBoey) [inline bullmq-task-queue/app/entry.client.tsx:4]: Could you please make sure we have an empty last line here?

```suggestion
hydrate(<RemixBrowser />, document);

```

**Review by MichaelDeBoey — CHANGES_REQUESTED** [review]

File: bullmq-task-queue/app/entry.client.tsx (hunk 7 above)
Comment (yaanno) [reply bullmq-task-queue/app/entry.client.tsx:4]: Ouch, I knew I need an other cup of coffee :) Sorry about this!

**Review by MichaelDeBoey — APPROVED** [review]

**Pull request merged**